File: main/tree-level-order-traversal/tree-level-order-traversal.cpp
/*
struct node
{
    int data;
    node* left;
    node* right;
}*/

#include <queue>

void LevelOrder(node* const root)
{
    if (root == nullptr) return;
    std::queue<node*> q;
    q.push(root);
    
    do {
        const auto n = q.front();
        q.pop();
        std::cout << n->data << ' ';
        
        if (n->left != nullptr) q.push(n->left);
        if (n->right != nullptr) q.push(n->right);
    }
    while (!q.empty());
}

File: main/tree-level-order-traversal/tree-level-order-traversal.java
	/* 
    
    class Node 
    	int data;
    	Node left;
    	Node right;
	*/
	public static void levelOrder(Node root) {
        if (root == null) return;
        
        final Deque<Node> queue = new ArrayDeque<>();
        queue.add(root);
        
        while (!queue.isEmpty()) {
            root = queue.remove();
            System.out.format("%d ", root.data);
            if (root.left != null) queue.add(root.left);
            if (root.right != null) queue.add(root.right);
        }
    }
    